bottlecap/src/lifecycle/invocation/context.rs

```diff
@@ -0,0 +1,117 @@
+use std::collections::VecDeque;
+
+use tracing::debug;
+
+#[derive(Debug, Clone)]
+pub struct Context {
+    pub request_id: String,
+    pub runtime_duration_ms: f64,
+    pub init_duration_ms: f64,
+    pub start_time: i64,
+}
+
+impl Context {
+    #[must_use]
+    pub fn new(
+        request_id: String,
+        runtime_duration_ms: f64,
+        init_duration_ms: f64,
+        start_time: i64,
+    ) -> Self {
+        Context {
+            request_id,
+            runtime_duration_ms,
+            init_duration_ms,
+            start_time,
+        }
+    }
+}
+
+#[allow(clippy::module_name_repetitions)]
+pub struct ContextBuffer {
```

Comment:
I decided to use the `ContextBuffer` so we can keep relevant data from the past `5` invocations in cache.

5 is just an arbitrary number, we can reduce/increase this number as desired.

Reply:
When a `PlatformRuntime` done hits, we effectively remove the `Context` in question, is this what you mean?

What do you mean by `drop invocation context which we haven't used for some reason yet`?